halo2_gadgets/src/ecc.rs

```diff
@@ -437,14 +437,13 @@ impl<C: CurveAffine, EccChip: EccInstructions<C> + Clone + Debug + Eq> Point<C,
 /// The affine short Weierstrass x-coordinate of a point on a specific elliptic curve.
 #[derive(Debug)]
 pub struct X<C: CurveAffine, EccChip: EccInstructions<C>> {
-    chip: EccChip,
     inner: EccChip::X,
 }
 
 impl<C: CurveAffine, EccChip: EccInstructions<C>> X<C, EccChip> {
     /// Wraps the given x-coordinate (obtained directly from an instruction) in a gadget.
-    pub fn from_inner(chip: EccChip, inner: EccChip::X) -> Self {
-        X { chip, inner }
+    pub fn from_inner(inner: EccChip::X) -> Self {
```

Comment:
What is the added value of that API stability, relative to the accuracy of the type signature?

Reply:
There are changes we may want to do in future that require ensuring that the chips of two gadgets are identical at runtime (to ensure they are operating over the same columns). We already do this in other places in the code. The type system check is insufficient because we may have two copies of a chip in the circuit.